app/src/main/java/de/westnordost/streetcomplete/view/image_select/GroupedImageSelectAdapter.kt

```diff
@@ -51,37 +75,42 @@ class GroupedImageSelectAdapter<T>(val gridLayoutManager: GridLayoutManager) :
 
     override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
         holder.bind(_items[position])
-        holder.isSelected = selectedItem?.let { _items.indexOf(it) == position } == true
-        holder.isGroupExpanded = getGroup(selectedIndex) == position
+        holder.isSelected = selectedItem?.let { _items.lastIndexOf(it) == position } == true
```

Comment:
There is still `lastIndexOf` here. Shouldn't it be

```suggestion
        holder.isSelected = selectedIndex() ==position
```

now?